app/assets/src/components/views/amr_heatmap/AMRHeatmapVis.jsx

```diff
@@ -52,8 +84,92 @@ export default class AMRHeatmapVis extends React.Component {
     return [sampleLabels, geneLabels, alleleLabels];
   }
 
+  //*** Callback functions for the heatmap ***
+
+  colorFilter = (value, node, originalColor, _, colorNoValue) => {
+    // Leave zero values grey
+    return value > 0 ? originalColor : colorNoValue;
+  };
+
+  onNodeHover = node => {
+    this.setState({ nodeHoverInfo: this.getTooltipData(node) });
+  };
+
+  onNodeHoverMove = (_, event) => {
+    if (event) {
+      this.setState({
+        tooltipLocation: {
+          left: event.pageX,
+          top: event.pageY,
+        },
+      });
+    }
+  };
+
+  onNodeHoverOut = () => {
+    this.setState({ nodeHoverInfo: null });
+  };
+
   //*** Following functions must be called after the component has updated ***
-  //*** (i.e. after the component has requested AMR data and updated state) ***
+
+  getTooltipData(node) {
+    const { samplesWithAMRCounts, selectedOptions } = this.props;
+    const {
+      sampleLabels,
+      geneLabels,
+      alleleLabels,
+      alleleToGeneMap,
+    } = this.state;
+    const sampleName = sampleLabels[node.columnIndex].label;
+    let amrCountIdentifier = "";
+    let gene = "";
+    let allele = "";
```

Comment:
Readability nit:
`amrCountIdentifier` feels a little vague. I wonder if it makes sense to name it "rowLabel".

You could do
`const rowLabel = this.getHeatmapLabels()[node.rowIndex].label`
before the `switch` statement, reusing your `getHeatmapLabels` function. (using getHeatmapLabels also makes the `rowLabel` name make more sense).

You can then also move the `sampleForColumn` and `amrCountForRow` assignments above the `switch` statement, and just assign `allele` and `gene` in the `switch` statement. This removes the need for the `if (gene === "")` conditions later on, which feel a little awkward.

All of this is to make the code a bit more readable.

Reply:
That's much more sensible, thank you! I've changed the code to incorporate your suggestion.